src/vs/code/node/cli.ts

```diff
@@ -242,6 +242,22 @@ export async function main(argv: string[]): Promise<any> {
 			});
 		}
 
+		// Handle --disable-state option
+		if (args['disable-state']) {
+			const tempUserDataDir = randomPath(tmpdir(), 'vscode-no-state-userdata');
+			const tempExtensionsDir = randomPath(tmpdir(), 'vscode-no-state-extensions');
+
```

Comment:
use the same parent folder for both so that there is only 1 random path for both